include/openthread/dataset.h

```diff
@@ -213,6 +213,17 @@ typedef struct otOperationalDatasetComponents
     bool mIsChannelMaskPresent : 1;      ///< TRUE if Channel Mask is present, FALSE otherwise.
 } otOperationalDatasetComponents;
 
+/**
+ * This structure represents a Thread Dataset timestamp.
+ *
+ */
+typedef struct otTimestamp
+{
+    uint64_t mSeconds : 48;
+    uint64_t mTicks : 15;
+    uint64_t mAuthoritative : 1;
```

Comment:
Why not `uint16_t` or `bool` for the other member variables?

Also I think it may help with code size if we don't use bit-fields here (not sure if we care to use less memory for this `struct`)? thoughts?

Reply:
Updated to use `uint16_t` and removed bit fields. This truly helps reducing the code size, see https://github.com/openthread/openthread/pull/7739#issuecomment-1140436004